rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/core/RocketMQTemplate.java

```diff
@@ -76,6 +80,179 @@ public void setMessageQueueSelector(MessageQueueSelector messageQueueSelector) {
+        if (Objects.isNull(message) || Objects.isNull(message.getPayload())) {
+            log.error("send request message failed. destination:{}, message is null ", destination);
+            throw new IllegalArgumentException("`message` and `message.payload` cannot be null");
+        }
+
+        try {
+            org.apache.rocketmq.common.message.Message rocketMsg = RocketMQUtil.convertToRocketMessage(rocketMQMessageConverter.getMessageConverter(),
+                charset, destination, message);
+            if (delayLevel > 0) {
+                rocketMsg.setDelayTimeLevel(delayLevel);
+            }
+            producer.request(rocketMsg, requestCallback, timeout);
+        } catch (Exception e) {
+            log.error("send request message failed. destination:{}, message:{} ", destination, message);
+            throw new MessagingException(e.getMessage(), e);
+        }
+    }
+
+    public void requestAsync(String destination, Object payload, RequestCallback requestCallback, long timeout, int delayLevel) {
+        Message<?> message = MessageBuilder.withPayload(payload).build();
+        requestAsync(destination, message, requestCallback, producer.getSendMsgTimeout(), delayLevel);
+    }
+
+    public void requestAsyncOrderly(String destination, Message<?> message, RequestCallback requestCallback, String hashKey) {
+        requestAsyncOrderly(destination, message, requestCallback, hashKey, producer.getSendMsgTimeout());
+    }
+
+    public void requestAsyncOrderly(String destination, Object payload, RequestCallback requestCallback, String hashKey) {
+        requestAsyncOrderly(destination, payload, requestCallback, hashKey, producer.getSendMsgTimeout());
+    }
+
+    public void requestAsyncOrderly(String destination, Message<?> message, RequestCallback requestCallback, String hashKey, long timeout) {
+        requestAsyncOrderly(destination, message, requestCallback, hashKey, timeout, 0);
+    }
+
+    public void requestAsyncOrderly(String destination, Object payload, RequestCallback requestCallback, String hashKey, long timeout) {
+        requestAsyncOrderly(destination, payload, requestCallback, hashKey, timeout, 0);
+    }
+
+    public void requestAsyncOrderly(String destination, Message<?> message, RequestCallback requestCallback, String hashKey, long timeout, int delayLevel) {
+        if (Objects.isNull(message) || Objects.isNull(message.getPayload())) {
+            log.error("send request message failed. destination:{}, message is null ", destination);
+            throw new IllegalArgumentException("`message` and `message.payload` cannot be null");
+        }
+
+        try {
+            org.apache.rocketmq.common.message.Message rocketMsg = RocketMQUtil.convertToRocketMessage(rocketMQMessageConverter.getMessageConverter(),
+                charset, destination, message);
+            if (delayLevel > 0) {
+                rocketMsg.setDelayTimeLevel(delayLevel);
+            }
```

Comment:
In the this method, there is need to check requestCallback is null?

Reply:
I think null is allowed here because we may have the situation that do nothing when response comes back.